prog-template.c

```diff
@@ -193,27 +198,43 @@ bool rgb_2_gray_scale(unsigned char* original, unsigned char* result) {
     return true;
 }   
 
-bool processImageFrame(unsigned char* buffer, apriltag_detector_t *td) {
-
+bool processImageFrame(unsigned char* buffer, apriltag_detector_t *td, apriltag_detection_t *det) {
     int result = false;
     image_u8_t im = { .width = IMG_WIDTH, .height = IMG_HEIGHT, .stride = IMG_WIDTH, .buf = buffer };
 
     zarray_t *detections = apriltag_detector_detect(td, &im);
-    int i;
-    for (i = 0; i < zarray_size(detections); i++) {
-        apriltag_detection_t *det;
-        zarray_get(detections, i, &det);
-
-        printf("detection %3d: id (%2dx%2d)-%-4d, hamming %d, margin %8.3f\n",
-                           i, det->family->nbits, det->family->h, det->id, det->hamming, det->decision_margin);
-    //     // Do stuff with detections here.
-
-        result = true;
+    if(detections)
+    {
+
+        int i;
+        for (i = 0; i < zarray_size(detections); i++) {
+            zarray_get(detections, i, &det);
+            printf("detection %3d: id (%2dx%2d)-%-4d, hamming %d, margin %8.3f\n",
+                            i, det->family->nbits, det->family->h, det->id, det->hamming, det->decision_margin);
+            result = true;
+            if(result)
```

Comment:
Don't need if statement here; result is set to true right before it

Reply:
Done